mc/test/timer.cpp

```diff
@@ -19,7 +19,7 @@ template <typename C>
 class TestCallback : public ::testing::Test {
   public:
     typedef C callback_type;
-    static const std::size_t TIMELIMIT=3; 
+    static const std::size_t TIMELIMIT=1;
```

Comment:
POSIX timer resolution is 1 sec. I am not sure that running simulation for 1 second is good enough test. Maybe 2 seconds? (and, does reducing test time by 2 seconds noticeably affects the total test run time?)